gammapy/spectrum/flux_point.py

```diff
@@ -1129,3 +1129,141 @@ def likelihood(self, parameters, mask=None):
             stat = self.likelihood_per_bin()[mask & self.mask]
 
         return np.nansum(stat, dtype=np.float64)
+
+    def residuals(self):
+        """Compute flux point residuals
+
+        Defined as `(data - model) / model`.
+
+        Returns
+        -------
+        residuals : `~numpy.ndarray`
+            Flux point residuals
+        """
+        fp = self.data
+        data = fp.table[fp.sed_type].quantity
+
+        model = self.model(fp.e_ref)
+        residuals = ((data - model) / model).to_value("")
+
+        # Remove residuals for upper_limits
+        residuals[fp.is_ul] = np.nan
+        return residuals
+
+    def peek(self):
+        """Plot flux points, best fit model and residuals.
+        """
+        from matplotlib.gridspec import GridSpec
+        import matplotlib.pyplot as plt
+
+        gs = GridSpec(7, 1)
+
+        ax_spectrum = plt.subplot(gs[:5, :])
+        self.plot_spectrum(ax=ax_spectrum)
+
+        ax_spectrum.set_xticks([])
+
+        ax_residuals = plt.subplot(gs[5:, :])
+        self.plot_residuals(ax=ax_residuals)
+        return ax_spectrum, ax_residuals
+
+    @property
+    def _e_range(self):
+        return u.Quantity([self.data.e_min.min(), self.data.e_max.max()])
+
+    @property
+    def _e_unit(self):
+        return self.data.e_ref.unit
+
+    def plot_residuals(self, ax=None, **kwargs):
+        """Plot flux point residuals.
+
+        Parameters
+        ----------
+        ax : `~matplotlib.pyplot.Axes`
+            Axes object.
+        **kwargs : dict
+            Keyword arguments passed to `~matplotlib.pyplot.errorbar`.
+
+        Returns
+        -------
+        ax : `~matplotlib.pyplot.Axes`
+            Axes object.
+        """
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca() if ax is None else ax
+
+        residuals = self.residuals()
+
+        fp = self.data
+
+        xerr = fp._plot_get_energy_err()
+        xerr = xerr[0].to_value(self._e_unit), xerr[1].to_value(self._e_unit)
+
+        model = self.model(fp.e_ref)
+        yerr = fp._plot_get_flux_err(fp.sed_type)
+        yerr = (yerr[0] / model).to_value(""), (yerr[1] / model).to_value("")
+
+        kwargs.setdefault("marker", "+")
+        kwargs.setdefault("ls", "None")
+        ax.errorbar(self.data.e_ref.value, residuals, xerr=xerr, yerr=yerr, **kwargs)
+
+        # format axes
+        ax.axhline(0, color="black", lw=0.5)
+        ax.set_ylabel("Residuals")
+        ax.set_xlabel("Energy ({})".format(self._e_unit))
+        ax.set_ylim(-1, 1)
```

Comment:
Fix the `y` scale or leave it free to adapt to the data?

Reply:
I wasn't sure, but I thought it's better to set a reasonable default, so that the standard plot always looks the same. Users can change it by setting `ax.set_ylim()` on the return axes objects outside the plotting method anyway.